src/middlewared/middlewared/plugins/vm/supervisor/supervisor_linux.py

```diff
@@ -57,3 +57,29 @@ def devices_xml(self):
 
         devices.extend([create_element('serial', type='pty'), create_element('video')])
         return create_element('devices', attribute_dict={'children': devices})
+
+    def cpu_xml(self):
+        cpu_elem = super().cpu_xml()
+        if self.vm_data['cpu_mode']:
+            cpu_elem.set('mode', self.vm_data['cpu_mode'].lower())
+        elif self.vm_data['cpu_model']:
```

Comment:
Do we really need two fields and use features+vendor xml?

On libvirt if I choose Opteron_G1 all it does in XML is:

<cpu mode="custom" match="exact" check="none">
    <model fallback="forbid">Opteron_G1</model>
</cpu>

Reply:
The 2 fields serve different purposes.
`cpu_mode` defines how the CPU will be configured, would it be a custom model or host-passthrough/host-model.
`cpu_model` shows which CPU will be used if custom cpu mode is desired. 
So in my opinion we should not be mixing them by using a single field because they represent different things.

About the features + vendor xml, I am doing that for completeness, we could in the future allow control at a feature level allowing user to force/require somethings. I am not 100% sure if there would be any drawbacks if we don't specify this, but I think there shouldn't be. If you think we should remove that, please let me know and I can confirm and remove.

Reply:
Yes, I think we dont need features + vendor as that only complicates things for no real benefit.

If we can avoid reading ~65 files and running 65 subprocesses to get the models, even better.